src/Columns/ColumnNullable.cpp

```diff
@@ -137,18 +137,37 @@ void ColumnNullable::insertData(const char * pos, size_t length)
 StringRef ColumnNullable::serializeValueIntoArena(size_t n, Arena & arena, char const *& begin) const
 {
     const auto & arr = getNullMapData();
+    const bool is_null = arr[n];
     static constexpr auto s = sizeof(arr[0]);
+    char * pos;
+    if (const ColumnString * string_col = checkAndGetColumn<ColumnString>(getNestedColumn()))
+    {
+        auto data = string_col->getDataAt(n);
```

Comment:
am I right that the whole idea of optimisation is to devirtualize `serializeValueIntoArena`? if so I think it would be better to just `static_cast` to it and call `serializeValueIntoArena` as usual.

Reply:
I'm not quite sure that the key to this optimization is devirtualize serializeValueIntoArena, and I've tested that this optimization has performance degradation on the number type. So need to determine whether the nested column is a ColumnString

Reply:
![image](https://github.com/ClickHouse/ClickHouse/assets/16730247/908ffe63-bf57-40ec-b1ce-969e32cd1a4f)
I tested the above modification and there was no impact on performance

Reply:
> I'm not quite sure that the key to this optimization is devirtualize serializeValueIntoArena, and I've tested that this optimization has performance degradation on the number type. So need to determine whether the nested column is a ColumnString

There are some new discoveries that I will try again to apply in the fixed size column

Reply:
I would assume that calculating this condition for every element may hurt perf (especially for numeric types, because it is the fastest case). better would be to move this condition out of the hot loop

Reply:
@nickitat After my test, the performance improvement should be due to function inline and reduced allocContinue calls. If want to modify this part of the code to each column class, the amount of change will be very large

Reply:
> I would assume that calculating this condition for every element may hurt perf (especially for numeric types, because it is the fastest case). better would be to move this condition out of the hot loop

done

Reply:
I added two variables to ColumnNullable to represent the type of nested column. I thought about moving these two variables to HashMethodSerialized, but this optimization can only be used in aggregation, and other places where serializeIntoArea is used cannot obtain optimization effects. So I think it's more appropriate to place these two variables in ColumnNullable